spec/models/user_spec.rb

```diff
@@ -176,24 +176,77 @@
     end
   end
 
-  describe '.search' do
-    subject { User.search(params) }
+  describe '.filter' do
     let(:params) { {} }
 
-    before(:each) do
-      FactoryGirl.create(:user, first_name: 'Bob', created_at: 5.days.ago)
-      FactoryGirl.create(:user, first_name: 'Marley', created_at: 2.days.ago)
-      FactoryGirl.create(:user, first_name: 'Janice', display_profile: false)
+    context 'has filters' do
+      before(:each) do
+        @user1 = FactoryGirl.create(:user, longitude: 19.15)
+        @user2 = FactoryGirl.create(:user, longitude: 27.74)
+        @project = FactoryGirl.create(:project)
+      end
+
+      it 'filters users for project' do
+        @user1.follow @project
+        @user2.stop_following @project
+        params['project_filter'] = @project.id
+
+        results = User.filter(params.slice('project_filter')).allow_to_display
+        
+        expect(results).to include(@user1)
+        expect(results).not_to include(@user2)
+      end
+
+      context 'filters users for timezone area' do
+        before(:each) do
+          @current_user = FactoryGirl.create(:user, longitude: 19.15)
+        end
+        
+        it 'filters user1 when choose Close To My Timezone Area' do
+          params['timezone_filter'] = [@current_user.longitude - 7.5, @current_user.longitude + 7.5]
+
+          results = User.filter(params.slice('timezone_filter')).allow_to_display
+        
+          expect(results).to include(@user1)
+          expect(results).not_to include(@user2)
+        end
+
+        it 'filters both users when choose Wider Timezone Area' do
+          params['timezone_filter'] = [@current_user.longitude - 22.5, @current_user.longitude + 22.5]
+
+          results = User.filter(params.slice('timezone_filter')).allow_to_display
+        
+          expect(results).to include(@user1)
+          expect(results).to include(@user2)
+        end
+      end
+
+      it 'does not raise error when filters are empty' do
+        params['project_filter'] = ''
+        params['timezone_filter'] = ''
+
+        expect{ User.filter(params.slice('project_filter', 'timezone_filter')).allow_to_display }.to_not raise_error
+      end
     end
 
-    it 'should be ordered by creation date' do
-      expect(subject.first.first_name).to eq('Bob')
-    end
+    context 'no filters' do
+      subject { User.filter(params.slice('project_filter')).allow_to_display }
+
+      before(:each) do
+        FactoryGirl.create(:user, first_name: 'Bob', created_at: 5.days.ago)
+        FactoryGirl.create(:user, first_name: 'Marley', created_at: 2.days.ago)
+        FactoryGirl.create(:user, first_name: 'Janice', display_profile: false)
+      end
+
+      it 'should be ordered by creation date' do
+        expect(subject.first.first_name).to eq('Bob')
```

Comment:
the creator of rspec suggests not to use 'should' in favour of 'it orders by creation date'.  He mentions that he used should when introducing rspec to make people think the correct way.